libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/LanguageGenerationBotComponent.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.Bot.Builder.Dialogs.Adaptive.Templates;
+using Microsoft.Bot.Builder.Dialogs.Declarative;
+using Microsoft.Bot.Builder.Dialogs.Declarative.Converters;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Bot.Builder.Dialogs.Adaptive
+{
+    /// <summary>
+    /// ComponentRegistration class for language generation resources.
+    /// </summary>
+    public class LanguageGenerationBotComponent : BotComponent
+    {
+        /// <inheritdoc/>
+        public override void Startup(IServiceCollection services, IConfiguration configuration, ILogger logger)
```

Comment:
did you try an attribute rather than this inheritance model?

Reply:
Did try. An attribute add another layer of opt-in for component authors (just like a[sp.net does with IHostingProvider interface and HostingStartup attribute](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/host/platform-specific-configuration?view=aspnetcore-5.0)) but it does not eliminate the need for inheritance and / or implementation.